src/Conversion/ONNXToTOSA/Math/Gemm.cpp

```diff
@@ -188,19 +188,25 @@ class ONNXGemmOpLoweringToTOSA : public OpConversionPattern<ONNXGemmOp> {
     // If all check passed, we replace the GEMM by a FC operator
     Type resultType = getTypeConverter()->convertType(op.getResult().getType());
 
-    // If no bias is given to the GEMM operator, we create a 1D bias with all
-    // zeroes
-    if (C.getType().isa<mlir::NoneType>()) {
-      // Base C format on B[0] format
-      ArrayRef<int64_t> cformat(B.getType().cast<TensorType>().getShape()[0]);
-      std::vector<float> elements = {};
-      for (int i = 0; i < cformat[0]; ++i)
-        elements.push_back(0.0F);
-      C = tosa::getConstTensor<float>(rewriter, op, elements, cformat).value();
+    // Because the bias is not broadcastable for TOSA while it is for Onnx,
+    // we create an empty bias and use an add (broadcastable for tosa)
+    // afterwards
+    // Base dummy C shape on B[0] shape
+    ArrayRef<int64_t> cformat(B.getType().cast<TensorType>().getShape()[0]);
+    std::vector<float> elements = {};
+    for (int i = 0; i < cformat[0]; ++i)
+      elements.push_back(0.0F);
+    Value dummyC = tosa::getConstTensor<float>(rewriter, op, elements, cformat).value();
```

Comment:
Are you always creating the constant? I think for upstreaming it would be better to only use this scheme if the shape is 1 (and the correct shape is not 1 by coincidence).